buddy_alloc.h

```diff
@@ -352,7 +395,7 @@ size_t buddy_sizeof(size_t memory_size) {
         return 0; /* invalid */
     }
     size_t buddy_tree_order = buddy_tree_order_for_memory(memory_size);
-    return sizeof(struct buddy) + buddy_tree_sizeof(buddy_tree_order);
+    return sizeof(struct buddy) + buddy_tree_sizeof((uint8_t)buddy_tree_order);
```

Comment:
Maybe we should consider changing the return type in all functions that return tree order to remove all implicit casts.

Reply:
That's okay for now